luxonis_ml/data/datasets/base_dataset.py

```diff
@@ -0,0 +1,131 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, Generator, List, Optional, Tuple
+
+from luxonis_ml.utils.registry import AutoRegisterMeta, Registry
+
+from .source import LuxonisSource
+
+DATASETS_REGISTRY = Registry(name="datasets")
+
+DatasetGenerator = Generator[Dict[str, Any], None, None]
+DatasetGeneratorFunction = Callable[[], DatasetGenerator]
+
+
+class BaseDataset(ABC, metaclass=AutoRegisterMeta, register=False):
```

Comment:
the `registry=DATASETS_REGISTRY` argument can be also specified here in the base instead of later in `LuxonisDataset`.